Regression tester regressed on revision 434975

```
Test failures disrupt,
deal.II's reliability,
Swift action required.
```

https://cdash.dealii.org/index.php?project=deal.II&filtercount=1&showfilters=1&field1=revision&compare1=63&value1=434975

@drwells @bangerth In addition two tests regressed: https://cdash.dealii.org/viewTest.php?onlyfailed&buildid=547
```/home/testsuite/workspace/regression_tests/dealii/tests/fe/fe_enriched_step-36.cc: In instantiation of 'void Step36::EigenvalueProblem<dim>::estimate_error() [with int dim = 3]':
/home/testsuite/workspace/regression_tests/dealii/tests/fe/fe_enriched_step-36.cc:628:9:   required from 'void Step36::EigenvalueProblem<dim>::run() [with int dim = 3]'
/home/testsuite/workspace/regression_tests/dealii/tests/fe/fe_enriched_step-36.cc:652:19:   required from here
/home/testsuite/workspace/regression_tests/dealii/tests/fe/fe_enriched_step-36.cc:505:41: error: no matching function for call to 'dealii::KellyErrorEstimator<3, 3>::estimate(dealii::DoFHandler<3, 3>&, dealii::hp::QCollection<2>&, std::map<unsigned int, const dealii::Function<3>*, std::less<unsigned int>, std::allocator<std::pair<const unsigned int, const dealii::Function<3>*> > >, std::vector<const dealii::PETScWrappers::MPI::Vector*>&, std::vector<dealii::Vector<float>*>&)'
  505 |       KellyErrorEstimator<dim>::estimate(
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
  506 |         dof_handler,
      |         ~~~~~~~~~~~~                     
  507 |         face_quadrature_formula,
      |         ~~~~~~~~~~~~~~~~~~~~~~~~         
  508 |         std::map<types::boundary_id, const Function<dim> *>(),
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  509 |         sol,
      |         ~~~~                             
  510 |         error);
      |         ~~~~~~                           
In file included from /home/testsuite/workspace/regression_tests/dealii/tests/fe/fe_enriched_step-36.cc:52:
/home/testsuite/workspace/regression_tests/dealii/include/deal.II/numerics/error_estimator.h:341:3: note: candidate: 'template<class Number> static void dealii::KellyErrorEstimator<dim, spacedim>::estimate(const dealii::Mapping<dim, spacedim>&, const dealii::DoFHandler<dim, spacedim>&, const dealii::Quadrature<(dim - 1)>&, const std::map<unsigned int, const dealii::Function<spacedim, number>*>&, const dealii::ReadVector<Number>&, dealii::Vector<float>&, const dealii::ComponentMask&, const dealii::Function<spacedim>*, unsigned int, dealii::types::subdomain_id, dealii::types::material_id, Strategy) [with int dim = 3; int spacedim = 3]'
  341 |   estimate(
      |   ^~~~~~~~
/home/testsuite/workspace/regression_tests/dealii/include/deal.II/numerics/error_estimator.h:341:3: note:   template argument deduction/substitution failed:
/home/testsuite/workspace/regression_tests/dealii/tests/fe/fe_enriched_step-36.cc:505:41: note:   'std::vector<const dealii::PETScWrappers::MPI::Vector*>' is not derived from 'const std::map<unsigned int, const dealii::Function<3, number>*, std::less<unsigned int>, std::allocator<std::pair<const unsigned int, const dealii::Function<3, number>*> > >'
  505 |       KellyErrorEstimator<dim>::estimate(
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
  506 |         dof_handler,
      |         ~~~~~~~~~~~~                     
  507 |         face_quadrature_formula,
      |         ~~~~~~~~~~~~~~~~~~~~~~~~         
  508 |         std::map<types::boundary_id, const Function<dim> *>(),
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  509 |         sol,
      |         ~~~~                             
  510 |         error);
      |         ~~~~~~                           
/home/testsuite/workspace/regression_tests/dealii/include/deal.II/numerics/error_estimator.h:362:3: note: candidate: 'template<class Number> static void dealii::KellyErrorEstimator<dim, spacedim>::estimate(const dealii::DoFHandler<dim, spacedim>&, const dealii::Quadrature<(dim - 1)>&, const std::map<unsigned int, const dealii::Function<spacedim, number>*>&, const dealii::ReadVector<Number>&, dealii::Vector<float>&, const dealii::ComponentMask&, const dealii::Function<spacedim>*, unsigned int, dealii::types::subdomain_id, dealii::types::material_id, Strategy) [with int dim = 3; int spacedim = 3]'
  362 |   estimate(
      |   ^~~~~~~~
/home/testsuite/workspace/regression_tests/dealii/include/deal.II/numerics/error_estimator.h:362:3: note:   template argument deduction/substitution failed:
/home/testsuite/workspace/regression_tests/dealii/tests/fe/fe_enriched_step-36.cc:505:41: note:   'std::vector<const dealii::PETScWrappers::MPI::Vector*>' is not derived from 'const dealii::ReadVector<Number>'
  505 |       KellyErrorEstimator<dim>::estimate(
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
  506 |         dof_handler,
      |         ~~~~~~~~~~~~                     
  507 |         face_quadrature_formula,
      |         ~~~~~~~~~~~~~~~~~~~~~~~~         
  508 |         std::map<types::boundary_id, const Function<dim> *>(),
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  509 |         sol,
      |         ~~~~                             
  510 |         error);
      |         ~~~~~~                           
/home/testsuite/workspace/regression_tests/dealii/include/deal.II/numerics/error_estimator.h:391:3: note: candidate: 'template<class Number> static void dealii::KellyErrorEstimator<dim, spacedim>::estimate(const dealii::Mapping<dim, spacedim>&, const dealii::DoFHandler<dim, spacedim>&, const dealii::Quadrature<(dim - 1)>&, const std::map<unsigned int, const dealii::Function<spacedim, number>*>&, const std::vector<const dealii::ReadVector<Number>*>&, std::vector<dealii::Vector<float>*>&, const dealii::ComponentMask&, const dealii::Function<spacedim>*, unsigned int, dealii::types::subdomain_id, dealii::types::material_id, Strategy) [with int dim = 3; int spacedim = 3]'
  391 |   estimate(
      |   ^~~~~~~~
/home/testsuite/workspace/regression_tests/dealii/include/deal.II/numerics/error_estimator.h:391:3: note:   template argument deduction/substitution failed:
/home/testsuite/workspace/regression_tests/dealii/tests/fe/fe_enriched_step-36.cc:505:41: note:   'std::vector<const dealii::PETScWrappers::MPI::Vector*>' is not derived from 'const std::map<unsigned int, const dealii::Function<3, number>*, std::less<unsigned int>, std::allocator<std::pair<const unsigned int, const dealii::Function<3, number>*> > >'
  505 |       KellyErrorEstimator<dim>::estimate(
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
  506 |         dof_handler,
      |         ~~~~~~~~~~~~                     
  507 |         face_quadrature_formula,
      |         ~~~~~~~~~~~~~~~~~~~~~~~~         
  508 |         std::map<types::boundary_id, const Function<dim> *>(),
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  509 |         sol,
      |         ~~~~                             
  510 |         error);
      |         ~~~~~~                           
/home/testsuite/workspace/regression_tests/dealii/include/deal.II/numerics/error_estimator.h:412:3: note: candidate: 'template<class Number> static void dealii::KellyErrorEstimator<dim, spacedim>::estimate(const dealii::DoFHandler<dim, spacedim>&, const dealii::Quadrature<(dim - 1)>&, const std::map<unsigned int, const dealii::Function<spacedim, number>*>&, const std::vector<const dealii::ReadVector<Number>*>&, std::vector<dealii::Vector<float>*>&, const dealii::ComponentMask&, const dealii::Function<spacedim>*, unsigned int, dealii::types::subdomain_id, dealii::types::material_id, Strategy) [with int dim = 3; int spacedim = 3]'
  412 |   estimate(
      |   ^~~~~~~~
/home/testsuite/workspace/regression_tests/dealii/include/deal.II/numerics/error_estimator.h:412:3: note:   template argument deduction/substitution failed:
/home/testsuite/workspace/regression_tests/dealii/tests/fe/fe_enriched_step-36.cc:505:41: note:   mismatched types 'const dealii::ReadVector<Number>' and 'const dealii::PETScWrappers::MPI::Vector'
  505 |       KellyErrorEstimator<dim>::estimate(
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
  506 |         dof_handler,
      |         ~~~~~~~~~~~~                     
  507 |         face_quadrature_formula,
      |         ~~~~~~~~~~~~~~~~~~~~~~~~         
  508 |         std::map<types::boundary_id, const Function<dim> *>(),
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  509 |         sol,
      |         ~~~~                             
  510 |         error);
      |         ~~~~~~                           
/home/testsuite/workspace/regression_tests/dealii/include/deal.II/numerics/error_estimator.h:433:3: note: candidate: 'template<class Number> static void dealii::KellyErrorEstimator<dim, spacedim>::estimate(const dealii::Mapping<dim, spacedim>&, const dealii::DoFHandler<dim, spacedim>&, const dealii::hp::QCollection<(dim - 1)>&, const std::map<unsigned int, const dealii::Function<spacedim, number>*>&, const dealii::ReadVector<Number>&, dealii::Vector<float>&, const dealii::ComponentMask&, const dealii::Function<spacedim>*, unsigned int, dealii::types::subdomain_id, dealii::types::material_id, Strategy) [with int dim = 3; int spacedim = 3]'
  433 |   estimate(
      |   ^~~~~~~~
/home/testsuite/workspace/regression_tests/dealii/include/deal.II/numerics/error_estimator.h:433:3: note:   template argument deduction/substitution failed:
/home/testsuite/workspace/regression_tests/dealii/tests/fe/fe_enriched_step-36.cc:505:41: note:   'std::vector<const dealii::PETScWrappers::MPI::Vector*>' is not derived from 'const std::map<unsigned int, const dealii::Function<3, number>*, std::less<unsigned int>, std::allocator<std::pair<const unsigned int, const dealii::Function<3, number>*> > >'
  505 |       KellyErrorEstimator<dim>::estimate(
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
  506 |         dof_handler,
      |         ~~~~~~~~~~~~                     
  507 |         face_quadrature_formula,
      |         ~~~~~~~~~~~~~~~~~~~~~~~~         
  508 |         std::map<types::boundary_id, const Function<dim> *>(),
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  509 |         sol,
      |         ~~~~                             
  510 |         error);
      |         ~~~~~~                           
/home/testsuite/workspace/regression_tests/dealii/include/deal.II/numerics/error_estimator.h:455:3: note: candidate: 'template<class Number> static void dealii::KellyErrorEstimator<dim, spacedim>::estimate(const dealii::DoFHandler<dim, spacedim>&, const dealii::hp::QCollection<(dim - 1)>&, const std::map<unsigned int, const dealii::Function<spacedim, number>*>&, const dealii::ReadVector<Number>&, dealii::Vector<float>&, const dealii::ComponentMask&, const dealii::Function<spacedim>*, unsigned int, dealii::types::subdomain_id, dealii::types::material_id, Strategy) [with int dim = 3; int spacedim = 3]'
  455 |   estimate(
      |   ^~~~~~~~
/home/testsuite/workspace/regression_tests/dealii/include/deal.II/numerics/error_estimator.h:455:3: note:   template argument deduction/substitution failed:
/home/testsuite/workspace/regression_tests/dealii/tests/fe/fe_enriched_step-36.cc:505:41: note:   'std::vector<const dealii::PETScWrappers::MPI::Vector*>' is not derived from 'const dealii::ReadVector<Number>'
  505 |       KellyErrorEstimator<dim>::estimate(
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
  506 |         dof_handler,
      |         ~~~~~~~~~~~~                     
  507 |         face_quadrature_formula,
      |         ~~~~~~~~~~~~~~~~~~~~~~~~         
  508 |         std::map<types::boundary_id, const Function<dim> *>(),
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  509 |         sol,
      |         ~~~~                             
  510 |         error);
      |         ~~~~~~                           
/home/testsuite/workspace/regression_tests/dealii/include/deal.II/numerics/error_estimator.h:476:3: note: candidate: 'template<class Number> static void dealii::KellyErrorEstimator<dim, spacedim>::estimate(const dealii::Mapping<dim, spacedim>&, const dealii::DoFHandler<dim, spacedim>&, const dealii::hp::QCollection<(dim - 1)>&, const std::map<unsigned int, const dealii::Function<spacedim, number>*>&, const std::vector<const dealii::ReadVector<Number>*>&, std::vector<dealii::Vector<float>*>&, const dealii::ComponentMask&, const dealii::Function<spacedim>*, unsigned int, dealii::types::subdomain_id, dealii::types::material_id, Strategy) [with int dim = 3; int spacedim = 3]'
  476 |   estimate(
      |   ^~~~~~~~
/home/testsuite/workspace/regression_tests/dealii/include/deal.II/numerics/error_estimator.h:476:3: note:   template argument deduction/substitution failed:
/home/testsuite/workspace/regression_tests/dealii/tests/fe/fe_enriched_step-36.cc:505:41: note:   'std::vector<const dealii::PETScWrappers::MPI::Vector*>' is not derived from 'const std::map<unsigned int, const dealii::Function<3, number>*, std::less<unsigned int>, std::allocator<std::pair<const unsigned int, const dealii::Function<3, number>*> > >'
  505 |       KellyErrorEstimator<dim>::estimate(
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
  506 |         dof_handler,
      |         ~~~~~~~~~~~~                     
  507 |         face_quadrature_formula,
      |         ~~~~~~~~~~~~~~~~~~~~~~~~         
  508 |         std::map<types::boundary_id, const Function<dim> *>(),
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  509 |         sol,
      |         ~~~~                             
  510 |         error);
      |         ~~~~~~                           
/home/testsuite/workspace/regression_tests/dealii/include/deal.II/numerics/error_estimator.h:498:3: note: candidate: 'template<class Number> static void dealii::KellyErrorEstimator<dim, spacedim>::estimate(const dealii::DoFHandler<dim, spacedim>&, const dealii::hp::QCollection<(dim - 1)>&, const std::map<unsigned int, const dealii::Function<spacedim, number>*>&, const std::vector<const dealii::ReadVector<Number>*>&, std::vector<dealii::Vector<float>*>&, const dealii::ComponentMask&, const dealii::Function<spacedim>*, unsigned int, dealii::types::subdomain_id, dealii::types::material_id, Strategy) [with int dim = 3; int spacedim = 3]'
  498 |   estimate(
      |   ^~~~~~~~
/home/testsuite/workspace/regression_tests/dealii/include/deal.II/numerics/error_estimator.h:498:3: note:   template argument deduction/substitution failed:
/home/testsuite/workspace/regression_tests/dealii/tests/fe/fe_enriched_step-36.cc:505:41: note:   mismatched types 'const dealii::ReadVector<Number>' and 'const dealii::PETScWrappers::MPI::Vector'
  505 |       KellyErrorEstimator<dim>::estimate(
      |       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
  506 |         dof_handler,
      |         ~~~~~~~~~~~~                     
  507 |         face_quadrature_formula,
      |         ~~~~~~~~~~~~~~~~~~~~~~~~         
  508 |         std::map<types::boundary_id, const Function<dim> *>(),
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  509 |         sol,
      |         ~~~~                             
  510 |         error);
      |         ~~~~~~                           

```

--- a/include/deal.II/lac/trilinos_tpetra_vector.templates.h
+++ b/include/deal.II/lac/trilinos_tpetra_vector.templates.h
@@ -120,7 +120,7 @@ namespace LinearAlgebra
 
     template <typename Number>
     void
-    Vector::extract_subvector_to(
+    Vector<Number>::extract_subvector_to(
       const ArrayView<const types::global_dof_index> &indices,
       ArrayView<Number> &                             elements) const
     {

--- a/include/deal.II/lac/trilinos_tpetra_vector.templates.h
+++ b/include/deal.II/lac/trilinos_tpetra_vector.templates.h
@@ -126,7 +126,7 @@ namespace LinearAlgebra
     {
       AssertDimension(indices.size(), elements.size());
       const auto &vector = trilinos_vector();
-      const auto &map    = vector.Map();
+      const auto &map    = vector.getMap();
 
 #  if DEAL_II_TRILINOS_VERSION_GTE(13, 2, 0)
       auto vector_2d = vector.template getLocalView<Kokkos::HostSpace>(
@@ -140,7 +140,7 @@ namespace LinearAlgebra
       for (unsigned int i = 0; i < indices.size(); ++i)
         {
           AssertIndexRange(indices[i], size());
-          const trilinos_i = map->getLocalElement(
+          const auto trilinos_i = map->getLocalElement(
             static_cast<TrilinosWrappers::types::int_type>(indices[i]));
           elements[i] = vector_1d(trilinos_i);
         }

